### PR TITLE
docs(appium): move sponsors into appium.io page

### PR DESCRIPTION
**Pull request by KazuCocoa**
Trying: https://github.com/appium/appium/issues/20621

Then, this idea makes current sponsors page into appium/docs. Then, the log links in this github repo misses.

(my local http server)

Screenshot 2024-09-27 at 12 20 18 AM

Screenshot 2024-09-27 at 8 42 56 AM

I won't manage this page in two places e.g. does the same thing in both github/SPONSORS.md and packages/appium/docs/en/sponsors.md, so... then I thought managing it only in packages/appium/docs/en/sponsors.md would be better.

### CODE REVIEW

File: packages/appium/docs/en/sponsors.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
+hide:
+  - navigation
+  - toc
+
+title: Sponsors & Backers
+---
```
Comment (TDL-EdgarsEglitis) [inline packages/appium/docs/en/sponsors.md:7]: Suggest adding some CSS to hide the sponsors footer on this page, like on the main page:
```
<style>
  .appium-sponsor-thanks {
    display: none;
  }
</style>
```

Comment (eglitise) [inline packages/appium/docs/en/sponsors.md:7]: Suggest adding some CSS here to hide the sponsors footer, like on the main page:
```
<style>
  .appium-sponsor-thanks {
    display: none;
  }
</style>
```

File: packages/appium/docs/mkdocs-en.yml [hunk 2]
```diff
@@ -75,3 +75,5 @@ nav:
       - Developer Reference:
           - developing/config-system.md
   - Contributing: contributing.md
+  - Sponsors:
+      - sponsors.md
```
Comment (eglitise) [inline packages/appium/docs/mkdocs-en.yml:79]: This can be a single line, no need to add `sponsors.md` as a list item

**Comment by eglitise** [comment]
The link to sponsors page in the main README should also be changed.

File: packages/appium/docs/en/sponsors.md (hunk 1 above)
Comment (KazuCocoa) [reply packages/appium/docs/en/sponsors.md:7]: Updated. The description also has with this change

**Review by eglitise — APPROVED** [review]

**Review by jlipps — APPROVED** [review]

**Pull request merged**